pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -216,7 +216,12 @@ func (w *Watcher) handleCreateEvent(event fsnotify.Event) error {
 	}
 
 	if !fi.IsDir() {
-		return w.handlePluginRegistration(event.Name)
+		if fi.Mode()&os.ModeSocket != 0 {
+			return w.handlePluginRegistration(event.Name)
+		} else {
+			glog.V(5).Infof("Ignoring file: %s with mode %v", fi.Name(), fi.Mode())
```

Comment:
How often do we expect to hit this?  If we only get 1 create event per socket, we might want this to be more verbose

Reply:
> If we only get 1 create event per socket

Yes.

This piece of code is a prevention which has the same logic as `traversePluginDir`. If someone created a non-socket file, the origin function will report an error message like `dial failed at socket`, I think this error shouldn't be raised because non-socket files are not our concerns.